frontend/src/common/ReviewWithComment.tsx

```diff
@@ -83,10 +85,13 @@ export default function ReviewWithComment({
 
   function submitForm(decision: ResponseTypeKeys) {
     setShowError(false)
+    setLoading(false)
     if (invalidComment() && decision === ResponseTypes.RequestChanges) {
       setShowError(true)
+      setLoading(false)
     } else {
       onSubmit(decision, reviewComment, reviewRequest.role)
```

Comment:
You don't need all of these `setLoading` calls. We should manage the loading state in the same place we're interacting with the API. In this case that's not in this component. All this component does is tell the parent component that it wants to submit some data. It's then up to that parent component to decide what to do with that data.

So in this case `ReviewBanner` is taking this data and sending it off to the backend. That means you'll want to manage the loading state in `ReviewBanner` also.

I suggest you change `ReviewWithComment` to have a new prop called `loading`. This will just be a boolean that you can pass to your `LoadingButton`s. Then you can have `const [loading, setLoading] = useState(false)` and your `setLoading(true)` and `setLoading(false)` in `ReviewBanner`. You can then pass this `loading` state value from `ReviewBanner` into `ReviewWithComment`.